### PR TITLE
Remove songs from detail activities after they are deleted

### LINKED ISSUE #339: Songs remain in the album detail list after being deleted
**Describe the bug**

**To Reproduce**
1. In the library, go to the albums tab and select an album
2. Tap the vertical elipses beside one of the songs to open the menu
3. Select "Delete from device"
4. Tap Delete
5. The song that was just deleted remains in the list.

The same thing happens when selecting multiple songs and using the menu in the upper right to delete them.

**Expected behavior**
The songs that were deleted disappear from the list.

**Smartphone:**
 - Device: OnePlus 6
 - Android Version: 10
 - ROM: Stock (OxygenOS 10.3.6)
 - Vinyl Music Player Version: 1.0.0

**Additional context**
After the song is deleted, the cache is refreshed from the database:
https://github.com/AdrienPoupa/VinylMusicPlayer/blob/f5865a664843963cc506fe264a1d151a9266d491/app/src/main/java/com/poupa/vinylmusicplayer/discog/Discography.java#L244-L247

and the list is reloaded from the cache:
https://github.com/AdrienPoupa/VinylMusicPlayer/blob/f5865a664843963cc506fe264a1d151a9266d491/app/src/main/java/com/poupa/vinylmusicplayer/ui/activities/AlbumDetailActivity.java#L405-L409

These both happen asynchronously, so the UI tends to get its list of songs (including the ones that were just deleted) while Discography is still waiting for results from the database to update the cache with.

### PR DESCRIPTION
**Pull request by Iislsdum**
Fixes #339

### CODE REVIEW

**Comment by AdrienPoupa** [comment]
Looks good to me.  @soncaokim thoughts?

**Comment by soncaokim** [comment]
Hi, looks good to me

**Comment by AdrienPoupa** [comment]
@Iislsdum is this ready for merge?

**Comment by Iislsdum** [comment]
@AdrienPoupa I will implement the same solution for the artist, playlist, and genre detail activities this evening, then it will be ready.

**Comment by AdrienPoupa** [comment]
Cool, no rush :)

File: app/src/main/java/com/poupa/vinylmusicplayer/ui/activities/AlbumDetailActivity.java [hunk 1]
```diff
@@ -116,6 +118,8 @@
     private MaterialDialog wikiDialog;
     private LastFMRestClient lastFMRestClient;
 
+    private final WeakMethodReference<AlbumDetailActivity> onDiscographyChanged = new WeakMethodReference<>(this, AlbumDetailActivity::reload);
```
Comment (soncaokim) [inline app/src/main/java/com/poupa/vinylmusicplayer/ui/activities/AlbumDetailActivity.java:121]: Is there any chance to move this ref and the onStart/Stop/Restart implementation to the base class AbsSlidingMusicPanelActivity, i.e. avoid copy/paste?

Comment (soncaokim) [reply app/src/main/java/com/poupa/vinylmusicplayer/ui/activities/AlbumDetailActivity.java:121]: @Iislsdum any update on this please?

Comment (Iislsdum) [reply app/src/main/java/com/poupa/vinylmusicplayer/ui/activities/AlbumDetailActivity.java:121]: Resolved in 48476d1

**Review by soncaokim — APPROVED** [review]
Looks good to me, thanks.

**Comment by AdrienPoupa** [comment]
Thanks!

**Pull request merged**